src/main/java/uk/gov/hmcts/reform/sandl/snlrules/model/Problem.java

```diff
@@ -5,17 +5,34 @@
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 import lombok.ToString;
+import org.apache.commons.codec.digest.DigestUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 @Getter
 @Setter
-@ToString(callSuper = true)
+@ToString
 @AllArgsConstructor
 @NoArgsConstructor
 public class Problem extends Fact {
     private String id;
     private String message;
+    private ProblemTypes type;
+    private ProblemSeverities severity;
+    private List<ProblemReference> references = new ArrayList<>();
+
+    public Problem(ProblemTypes type, ProblemSeverities severity, ProblemReference... references) {
+        this.type = type;
+        this.severity = severity;
+        this.references.addAll(Arrays.asList(references));
+
+        this.id = DigestUtils.md5Hex(this.toString());
```

Comment:
Why md5Hex? Can't we just use UUID?

Reply:
nope, problem id needs to be deterministic, so it cannot be randomly generated by rules, otherwise we will mess up database and would never know what problems are true. Therefore we use md5 hash for entire class data so it is always the same if only the data in the problem are the same.

if we reset rules and put again all the facts in it should produce exactly the same problems (including problems ids) which will happen with this md5 hash, but would not happen if uuid was created in rules as each time it would be different.

Reply:
Ok, I understand now 👍